exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/DbKeyCompressedFunnel.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.proofs;
+
+import com.exonum.binding.common.hash.Funnel;
+import com.exonum.binding.common.hash.PrimitiveSink;
+import com.exonum.binding.common.proofs.map.DbKey;
+
+/**
+ * A funnel for a database key. Puts the LEB128 compressed binary representation of the given
```

Comment:
The wording is misleading — LEB128 is not applied to the whole key, but to the key slice size only (or what is the current term of the size of the significant part of the key).

Reply:
Also we might say 'encodes' instead of compresses, as it 'compresses' anything greater than 127 into two bytes.